pkg/controller/kubelet-config/kubelet_config_nodes.go

```diff
@@ -270,8 +287,15 @@ func (ctrl *Controller) deleteNodeConfig(obj interface{}) {
 }
 
 func RunNodeConfigBootstrap(templateDir string, features *osev1.FeatureGate, cconfig *mcfgv1.ControllerConfig, nodeConfig *osev1.Node, mcpPools []*mcfgv1.MachineConfigPool) ([]*mcfgv1.MachineConfig, error) {
+	if nodeConfig == nil {
```

Comment:
For us to get into this function, there was an explicit `nodeConfig != nil` check, so maybe we don't need this?

Reply:
This function would also be called from the test framework to test the bootstrap behavior when a `nil` `nodeConfig` is sent. Hence, this check has been added. However in the real cluster scenario, we are defaulting the node object if found `nil`